gen/gen.go

```diff
@@ -110,32 +108,115 @@ func (g *Gen) Build(config *Config) error {
 		return err
 	}
 
-	swagger.Schemes = []string{}
-	b, err = json.MarshalIndent(swagger, "", "    ")
+	// Write doc
+	err = g.writeGoDoc(docs, swagger)
 	if err != nil {
 		return err
 	}
 
-	if err := packageTemplate.Execute(docs, struct {
-		Timestamp time.Time
-		Doc       string
-		Schemes   string
-	}{
-		Timestamp: time.Now(),
-		Doc:       "`" + strings.Replace(string(b), "{", "{\n    \"schemes\": {{ marshal .Schemes }},", 1) + "`",
-		Schemes:   "[]string{" + strings.Join(schemes, ",") + "}",
-	}); err != nil {
-		return err
-	}
-
 	log.Printf("create docs.go at  %+v", docs.Name())
 	log.Printf("create swagger.json at  %+v", swaggerJSON.Name())
 	log.Printf("create swagger.yaml at  %+v", swaggerYAML.Name())
 
 	return nil
 }
 
-var packageTemplate = template.Must(template.New("").Parse(`// GENERATED BY THE COMMAND ABOVE; DO NOT EDIT
+func (g *Gen) jsonIndent(data interface{}) ([]byte, error) {
+	return json.MarshalIndent(data, "", "    ")
+}
+
+func (g *Gen) formatSource(src []byte) []byte {
+	code, err := format.Source(src)
+	if err != nil {
```

Comment:
why does return code instead of return `errr`?

Reply:
If formating goes wrong , mod.go is empty. You won't be able to see what is wrong. 
If you want other behavior please let me know and i will change it that way.